python/lsst/pipe/tasks/imageDifference.py

```diff
@@ -108,6 +107,10 @@ class ImageDifferenceConfig(pexConfig.Config):
         target=ImagePsfMatchTask,
         doc="Warp and PSF match template to exposure, then subtract",
     )
+    decorrelate = pexConfig.ConfigurableField(
+        target=ImageDecorrelationTask,
+        doc="Decorrelate image difference",
+    )
```

Comment:
Please add a comment noting that it will be ignored if doSubtract is False (other than affecting the default value of detection.thresholdValue, which I think should be fixed).

Also, a brief explanation of WHY you only decorrelate if doSubtract is true would be nice. I assume it is because there is no noise to decorrelate.